samcli/lib/config/samconfig.py

```diff
@@ -46,16 +40,18 @@ def __init__(self, config_dir, filename=None):
         """
         self.document = {}
         self.filepath = Path(config_dir, filename or self.get_default_file(config_dir=config_dir))
-        self.file_manager = self.FILE_MANAGER_MAPPER.get(self.filepath.suffix, None)
+        file_extension = self.filepath.suffix
+        self.file_manager = FILE_MANAGER_MAPPER.get(file_extension, None)
         if not self.file_manager:
             LOG.warning(
-                f"The config file extension '{self.filepath.suffix}' is not supported. "
-                f"Supported formats are: [{'|'.join(self.FILE_MANAGER_MAPPER.keys())}]"
+                f"The config file extension '{file_extension}' is not supported. "
+                f"Supported formats are: [{'|'.join(FILE_MANAGER_MAPPER.keys())}]"
             )
             raise SamConfigFileReadException(
                 f"The config file {self.filepath} uses an unsupported extension, and cannot be read."
             )
         self._read()
+        EventTracker.track_event("SamConfigFileExtension", file_extension)
```

Comment:
Could we add a simple unit test to check that the expected file extension with that event name is passed in?